vllm/config/model_arch.py

```diff
@@ -0,0 +1,145 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+import functools
+import json
+from dataclasses import field
+from typing import Any
+
+from pydantic import ConfigDict
+from pydantic.dataclasses import dataclass
+from torch import nn
+
+from vllm.config.utils import config
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
+
+
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True, extra="allow"))
+class ModelArchitectureTextConfig:
+    model_type: str
+    hidden_size: int
+    num_hidden_layers: int
+    num_attention_heads: int
+    use_deepseek_mla: bool
+    head_dim: int
+    vocab_size: int
+    num_key_value_heads: int
```

Comment:
A model can have mixed different attention layers. Should we make some of the fields here per-layer?